src/librustc_parse/lib.rs

```diff
@@ -321,7 +321,12 @@ pub fn nt_to_tokenstream(nt: &Nonterminal, sess: &ParseSess, span: Span) -> Toke
             Some(tokenstream::TokenTree::token(token::Lifetime(ident.name), ident.span).into())
         }
         Nonterminal::NtTT(ref tt) => Some(tt.clone().into()),
-        _ => None,
+        Nonterminal::NtLiteral(ref expr) | Nonterminal::NtExpr(ref expr) => expr.tokens.clone(),
```

Comment:
This needs to use `prepend_attrs` in the same way as `Nonterminal::NtItem` uses it above.